master/buildbot/test/unit/www/test_avatar.py

```diff
@@ -663,3 +663,44 @@ def test_email_error(self):
                 'application/vnd.github.cloak-preview'})
         res = yield self.render_resource(self.rsrc, b'/?email=error@defunkt.com')
         self.assertEqual(res, dict(redirected=b'img/nobody.png'))
+
+
+class GitHubAvatarBasicAuth(TestReactorMixin, www.WwwTestMixin, unittest.TestCase):
+
+    @defer.inlineCallbacks
+    def setUp(self):
+        self.setUpTestReactor()
+
+        master = self.make_master(
+            url='http://a/b/', auth=auth.NoAuth(),
+            avatar_methods=[avatar.AvatarGitHub(
+                client_id="oauth_id", client_secret="oauth_secret")])
+
+        self.rsrc = avatar.AvatarResource(master)
+        self.rsrc.reconfigResource(master.config)
+
+        headers = {
+            'User-Agent': 'Buildbot',
+            # oauth_id:oauth_secret in Base64
+            'Authorization': 'basic b2F1dGhfaWQ6b2F1dGhfc2VjcmV0',
+        }
+        self._http = yield fakehttpclientservice.HTTPClientService.getService(
+            master, self,
+            avatar.AvatarGitHub.DEFAULT_GITHUB_API_URL,
+            headers=headers,
+            debug=False, verify=False)
+        yield self.master.startService()
+
+    @defer.inlineCallbacks
+    def tearDown(self):
+        yield self.master.stopService()
+
+    @defer.inlineCallbacks
+    def test_username(self):
+        username_search_endpoint = '/users/defunkt'
+        self._http.expect('get', username_search_endpoint,
+            content_json=github_username_search_reply,
+            headers={'Accept': 'application/vnd.github.v3+json'})
+        res = yield self.render_resource(self.rsrc, b'/?username=defunkt')
+        self.assertEqual(res, dict(redirected=b'https://avatars3.githubusercontent.com/'
```

Comment:
Let's use dict literal here.

Reply:
That's a copy paste from the classes above (test_gravatar, and GitHubAvatar).
Should I change these as well?

Reply:
No, let's move to the literal syntax gradually. Changing for the sake of changing will make git blame much less useful.

Reply:
Done